lib/galaxy/app.py

```diff
@@ -331,6 +331,8 @@ def _configure_toolbox(self):
             mulled_resolution_cache = CacheManager(**parse_cache_config_options(cache_opts)).get_cache(
                 "mulled_resolution"
             )
+            # If using database cache clear cache table contents
```

Comment:
It should be persistent across processes and restarts, what was the reason to change this ?

Reply:
I guess I don't know the requirements for these caches. How long are entries expected to stay in the caches? I made this change based upon your comment in the meeting that you didn't want the default cache type to be database because you were concerned that data would accumulate in the cache table and not be cleaned out. I also assumed that in a production environment app restarts would be very infrequent. Is that not the case? So I figured this would be a good place to clean out the table to address your concern. I presume data accumulation would also be a problem with a file based cache. So why is there a concern specifically with the database cache?

Reply:
Restarts can be frequent, for config updates, or because of resources limits, and they can be simultaneous or staggered, so I don't think this is safe.

It's not specific to the database cache, but it's safe to assume people know how to delete a file, while dropping the appropriate table requires a bit more knowledge. The lifetime is typically unlimited, but we may have to drop data occasionally, that for instance was necessary when admins upgraded python versions with incompatible pickle protocols.

Reply:
That's a long answer to say that I would default to a file-based cache for simplicity reasons as you're doing now, and database is a good option for production deployments with kubernetes, where you can run out of (very low) file lock limits. If the cache needs to be cleared admins have to do this manually, and that's easier with files.